script/config.tlu

```diff
@@ -431,15 +431,35 @@ end
 -- doesn't work on windows (uses io.popen)
 -- logic stolen from xdg-open (http://www.freedesktop.org/) and adapted
 function desktop_environment_viewer()
-    if os.getenv('KDE_SESSION_VERSION') and is_in_path('kde-open') then
-        return '(kde-open %s) &'                    -- kde 4 (or greater)
+    kde_desktop_viewer = nil
+    if is_in_path('kde-open') then
+        kde_desktop_viewer = '(kde-open %s) &'
+    elseif is_in_path('kfmclient') then
+        kde_desktop_viewer = '(kfmclient exec %s) &'
     end
-    if os.getenv('KDE_FULL_SESSION') and is_in_path('kfmclient') then
-        return '(kfmclient exec %s) &'              -- kde < 4
+
+    gnome_desktop_viewer = nil
+    if is_in_path('gvfs-open') then
+        gnome_desktop_viewer = '(gvfs-open %s) &'
+    elseif is_in_path('gnome-open') then
+        gnome_desktop_viewer = '(gnome-open %s) &'
+    end
+
+    if (os.getenv('KDE_SESSION_VERSION') or os.getenv('KDE_FULL_SESSION'))
+        and kde_desktop_viewer then
+        return kde_desktop_viewer
     end
-    if os.getenv('GNOME_DESKTOP_SESSION_ID') then   -- gnome
-        if is_in_path('gvfs-open') then return '(gvfs-open %s) &' end
-        if is_in_path('gnome-open') then return '(gnome-open %s) &' end
+    if os.getenv('GNOME_DESKTOP_SESSION_ID') and gnome_desktop_viewer then   -- gnome
```

Comment:
I would keep this part, not everyone necessarily is running the newest gnome. We can check for either of these env vars set.

Reply:
This PR does not drop support of `$GNOME_DESKTOP_SESSION_ID`; `$XDG_CURRENT_DESKTOP` is used only when `$KDE_SESSION_VERSION`/`$KDE_FULL_SESSION` and `$GNOME_DESKTOP_SESSION_ID` are not set.

Reply:
Ok, thanks for the update